### ISSUE TITLE
Regarding softmax layer 

### ISSUE THREAD

**Issue by iitbombombay** [jiecaoyu/XNOR-Net-PyTorch#110]
hi @jiecaoyu , I wanted to know for last layer is there any softmax layer in  Imagenet (alexnet model)?,your help will be appreciated

**Comment by jiecaoyu** [comment]
@iitbombombay Yes, the softmax function is included in [CrossEntropyLoss](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss).

**Comment by iitbombombay** [comment]
Does it involved in classification(like prediction of classes) in last layer?

**Comment by iitbombombay** [comment]
hi @jiecaoyu , As you said softmax function is included in loss,I have inference code where i am not calculating loss there
I have modified your code for inference by looking into this https://discuss.pytorch.org/t/how-to-load-images-for-inference-in-batch/33329.
i have to add loss function there for including softmax function or is it fine ?
this is my code for inference:
```print('==> Using Pytorch Dataset')
    import torchvision
    import torchvision.transforms as transforms
    import torchvision.datasets as datasets
    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],
                std=[1./255., 1./255., 1./255.])

    torchvision.set_image_backend('PIL')

    data_transforms = {
    'predict': transforms.Compose([
        transforms.Resize((256, 256)),
        transforms.CenterCrop(input_size),
        transforms.ToTensor(),
        normalize,
        ])
    }

    dataset = {'predict' : datasets.ImageFolder("/content/XNOR-Net-PyTorch/ImageNet/networks/", data_transforms['predict'])}
    dataloader = {'predict': torch.utils.data.DataLoader(dataset['predict'], batch_size=args.batch_size, shuffle=False, num_workers=args.workers,pin_memory=True)}
batch_time = AverageMeter()
    losses = AverageMeter()
    top1 = AverageMeter()
    top5 = AverageMeter()

    model.eval()
    end = time.time()
    global bin_op
    bin_op = util.BinOp(model)
    bin_op.binarization()
    #input=img

    for input, labels in dataloader['predict']:
      with torch.no_grad():
        input_var = torch.autograd.Variable(input)
        
        
    # compute output
      output = model(input_var)
      values,indices=torch.max(output,1)```

**Comment by jiecaoyu** [comment]
@iitbombombay sorry I cannot quite understand your code. If you need to add softmax, you can use [```torch.nn.functional.softmax```](https://pytorch.org/docs/stable/nn.functional.html#softmax).
